Sources/Basics/Sandbox.swift

```diff
@@ -114,6 +113,15 @@ fileprivate func macOSSandboxProfile(
         contents += ")\n"
     }
 
+    // Emit rules for paths under which writing is allowed, even if they are descendants directories that are otherwise read-only.
+    if writableDirectories.count > 0 {
```

Comment:
there seems to be duplicate code block that write these out. is one redundant?

Reply:
They look almost the same, but the L99 one is emitting the expressions for the implicit writable directories, which includes both temporary directories and the Clang cache directories for pre-5.3 compatibility (the former are subpaths and the latter regexes).  The problem was that the explicitly specified writable directories were emitted as part of that same batch of directories, followed by the explicitly writable ones.  But really the correct order is:
1. implicitly writable (temporary directories and the Clang cache directories)
2. explicitly readable
3. explicitly writable

In the new PR https://github.com/apple/swift-package-manager/pull/4005 this is made a lot more clear with a bit less redundancy.